docs/_docs/reference/contextual/context-bounds.md

```diff
@@ -4,50 +4,205 @@ title: "Context Bounds"
+```scala
+  trait SemiGroup[A]:
+    extension (x: A) def combine(y: A): A
+
+  trait Monoid[A] extends SemiGroup[A]:
+    def unit: A
+```
+We can write `reduce` function over lists of monoid instances like this:
 ```scala
-def f[T, U](x: T)(using _: C1[T], _: C2[T], _: C3[U], y: U, z: V): R
+  def reduce[A: Monoid as m](xs: List[A]): A =
+    xs.foldLeft(m.unit)(_ `combine` _)
 ```
+We use `as x` after the type of a context bound to bind the instance to `x`. This is analogous to import renaming, which also introduces a new name for something that comes before.
 
-Context bounds can be combined with subtype bounds. If both are present, subtype bounds come first, e.g.
+In a context bound with a naming clause the witness parameter carries the given name. For instance the expanded signature of `reduce` would be
+```scala
+  def reduce[A](xs: List[A])(using m: Monoid[A]): A
+```
+Since the context parameter now has a name, it can be referred
+to in the body of `reduce`. An example is the `m.unit` reference in the definition above.
+
+If the context bound does not carry an `as` clause, the generated witness parameter gets a compiler-synthesized name. However, a [currently experimental
+language extension](../experimental/default-names-context-bounds.md) would in this case give the context parameter the same name as the bound type parameter.
+
+Named context bounds were introduced in Scala 3.6.
+
+## Aggregate Context Bounds
 
+A type parameter can have several context bounds. If there are multiple bounds, they are written inside braces `{...}`. Example:
 ```scala
-def g[T <: B : C](x: T): R = ...
+  trait:
+    def showMax[X : {Ord, Show}](x: X, y: X): String
+  class B extends A:
+    def showMax[X : {Ord as ordering, Show as show}](x: X, y: X): String =
+      show.asString(ordering.max(x, y))
 ```
 
-## Migration
+This syntax is valid from Scala 3.6. The previous syntax used
+chains of `:` clauses, as in `[X : Ord : Show]`. This syntax is still available but will be deprecated and removed over time.
 
-To ease migration, context bounds in Dotty map in Scala 3.0 to old-style implicit parameters
-for which arguments can be passed either with a `(using ...)` clause or with a normal application. From Scala 3.1 on, they will map to context parameters instead, as is described above.
+## Placement of Generated Context Parameters
 
-If the source version is `future-migration`, any pairing of an evidence
+The witness context parameter(s) generated from context bounds are added as follows:
+
+ 1. If one of the bounds is referred to by its name in a subsequent parameter clause, the context bounds are mapped to a using clause immediately preceding the first such parameter clause.
+ 2. Otherwise, if the last parameter clause is a using (or implicit) clause, merge all parameters arising from context bounds in front of that clause, creating a single using clause.
+ 3. Otherwise, let the parameters arising from context bounds form a new using clause at the end.
```

Comment:
The phrasing of this sentence sounds odd. Shouldn't it:
```suggestion
 3. Otherwise, the parameters arising from context bounds form a new using clause at the end.
```